docs/usage/dataclasses.md

```diff
@@ -126,11 +134,12 @@ class MyDataclass2:
 
 1. You can read more about `validate_assignment` in [model_config](model_config.md#validate-assignment).
 
-!!! warning
-    After v1.10, _pydantic_ dataclasses support `Config.extra` but some default behaviour of stdlib dataclasses
-    may prevail. For example, when `print`ing a _pydantic_ dataclass with allowed extra fields, it will still
-    use the `__str__` method of stdlib dataclass and show only the required fields.
-    This may be improved further in the future.
+!!! note
+    Pydantic dataclasses do not support [`extra='allow'`](model_config/#extra-attributes), where extra fields passed
+    to the initializer would be stored as extra attributes on the dataclass.
```

Comment:
Hi @tpdorsey! Could you help me understand what you meant by this diff? It looks to me like the current support for `extra="allow"` is the same as it was after #2557 (the PR where the removed copy was introduced), but I'm wondering if I'm missing something.

With `extra="allow"`, extra values passed to `__init__` still end up in the instantiated object, but just aren't surfaced via `__str__`. Example script:

```python
from pydantic.dataclasses import dataclass

@dataclass(config=dict(extra="allow"))
class Example:
    ...

obj = Example(a=1, b=2)
print(obj)
print(f"Extra: a={obj.a}, b={obj.b}")
```

This script yields the following output when running with the latest `pydantic==2.3.0`:

```
Example()
Extra: a=1, b=2
```

Reply:
Thanks for the quick response, @hramezani — opened https://github.com/pydantic/pydantic/discussions/7362 to discuss.